src/main/java/com/hubspot/jinjava/lib/filter/AbstractFilter.java

```diff
@@ -0,0 +1,238 @@
+      value == null ||
+      Arrays.asList("object", "dict", "sequence").contains(jinjavaParamMetadata.type())
+    ) {
+      return value;
+    }
+    String valueString = Objects.toString(value, null);
+    switch (jinjavaParamMetadata.type().toLowerCase()) {
+      case "boolean":
+        return value instanceof Boolean
+          ? (Boolean) value
+          : BooleanUtils.toBooleanObject(valueString);
+      case "int":
+        return value instanceof Integer
+          ? (Integer) value
+          : NumberUtils.toInt(valueString);
+      case "long":
+        return value instanceof Long ? (Long) value : NumberUtils.toLong(valueString);
+      case "float":
+        return value instanceof Float ? (Float) value : NumberUtils.toFloat(valueString);
+      case "double":
+        return value instanceof Double
+          ? (Double) value
+          : NumberUtils.toDouble(valueString);
+      case "number":
+        return value instanceof Number ? (Number) value : new BigDecimal(valueString);
+      case "string":
+        return valueString;
+      default:
+        throw new InvalidInputException(
+          interpreter,
+          "INVALID_ARG_NAME",
+          String.format(
+            "Argument named '%s' with value '%s' cannot be parsed for filter %s",
+            jinjavaParamMetadata.value(),
+            value,
+            getName()
+          )
+        );
+    }
+  }
+
+  public void validateArgs(
+    JinjavaInterpreter interpreter,
+    Map<String, Object> parsedArgs
+  ) {
+    for (JinjavaParam jinjavaParam : namedArguments.values()) {
+      if (jinjavaParam.required() && !parsedArgs.containsKey(jinjavaParam.value())) {
+        throw new InvalidInputException(
+          interpreter,
+          "MISSING_REQUIRED_ARG",
+          String.format(
+            "Argument named '%s' is required but missing for filter %s",
+            jinjavaParam.value(),
+            getName()
+          )
+        );
+      }
+    }
+  }
+
+  public int getNamedArgumentPosition(String argName) {
+    return Optional
+      .ofNullable(namedArguments)
+      .map(Map::keySet)
+      .map(ArrayList::new)
+      .flatMap(argNames -> Optional.of(argNames.indexOf(argName)))
+      .orElse(-1);
+  }
+
+  public String getIndexedArgumentName(int position) {
+    return Optional
+      .ofNullable(namedArguments)
+      .map(Map::keySet)
+      .map(ArrayList::new)
+      .flatMap(
+        argNames ->
+          Optional.ofNullable(argNames.size() > position ? argNames.get(position) : null)
+      )
+      .orElse(null);
+  }
+
+  public Map<String, JinjavaParam> initNamedArguments() {
+    JinjavaDoc jinjavaDoc = this.getClass().getAnnotation(JinjavaDoc.class);
+    if (jinjavaDoc != null) {
+      Map<String, JinjavaParam> namedArgs = new LinkedHashMap<>();
+      for (JinjavaParam jinjavaParam : jinjavaDoc.params()) {
+        namedArgs.put(jinjavaParam.value(), jinjavaParam);
+      }
+      return Collections.unmodifiableMap(namedArgs);
```

Comment:
```suggestion
return jinjavaDoc.params().stream().collect(ImmutableMap.toImmutableMap(p -> p.value(), p -> p));
```

Reply:
Note that this won't preserve the order though. If you do need that, you could collect to a LinkedHashMap and then wrap the whole thing in `Collections.unmodifiableMap`

Reply:
The order is important, just to be sure I used forEachOrdered with Builder